src/main/java/com/app/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+import com.app.onlinebookstore.repository.OrderItemRepository;
+import com.app.onlinebookstore.repository.OrderRepository;
+import com.app.onlinebookstore.repository.ShoppingCartRepository;
+import com.app.onlinebookstore.service.OrderService;
+import com.app.onlinebookstore.service.UserService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final UserService userService;
+
+    @Override
+    public OrderDto createOrder(CreateOrderRequestDto requestDto, Authentication authentication) {
+        User currentUser = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository
+                 .findByUserId(currentUser.getId()).orElseThrow(
+                        () -> new EntityNotFoundException("Can't find shopping cart by user id "
+                                + currentUser.getId()));
+        Order order = new Order();
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Status.PENDING);
+        order.setUser(currentUser);
+        order.setTotal(shoppingCart.getCartItems().stream()
+                .map(this::getTotalPrice)
+                .reduce(BigDecimal::add).orElseThrow());
+        Order savedOrder = orderRepository.save(order);
+        order.setOrderItems(shoppingCart.getCartItems().stream()
+                .map(i -> orderItemMapper.cartItemToOrderItem(i, savedOrder))
+                .map(orderItemRepository::save)
+                .collect(Collectors.toSet()));
+        return setOrderItems(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderDto> findAllOrders(Pageable pageable) {
+        User user = userService.getAuthenticatedUser();
+        List<Order> orders = orderRepository.getOrderByUserId(user.getId());
+        for (Order order : orders) {
+            order.setOrderItems(orderItemRepository.getAllByOrderId(order.getId()));
+        }
+        return orders.stream()
+                .map(this::setOrderItems)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Set<OrderItemDto> findAllOrderItems(Long orderId) {
+        User user = userService.getAuthenticatedUser();
+        Set<OrderItem> orderItems = orderRepository.getOrderByUserId(user.getId()).stream()

```

Comment:
Please retrieve already assigned to user(order) orderItems from db, not by retrieving all and filtering. Please do same in other methods